python/az/aro/azext_aro/_dynamic_validators.py

```diff
@@ -0,0 +1,335 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the Apache License 2.0.
+
+import ipaddress
+import re
+from itertools import tee
+
+from azure.cli.core.commands.client_factory import get_mgmt_service_client
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from azure.cli.core.profiles import ResourceType
+from azure.cli.core.azclierror import CLIInternalError, InvalidArgumentValueError, \
+    RequiredArgumentMissingError
+from azure.core.exceptions import ResourceNotFoundError
+from knack.log import get_logger
+from msrestazure.tools import is_valid_resource_id
+from msrestazure.tools import parse_resource_id
+from msrestazure.azure_exceptions import CloudError
+from azext_aro._validators import validate_vnet, validate_cidr
+from azext_aro._rbac import has_role_assignment_on_resource
+import azext_aro.custom
+
+
+logger = get_logger(__name__)
+
+
+def can_do_action(perms, action):
+    for perm in perms:
+        for not_action in perm.not_actions:
+            match = re.escape(not_action)
+            match = re.match("(?i)^" + match.replace(r"\*", ".*") + "$", action)
+            if match:
+                return f"{action} permission is disabled"
+        for perm_action in perm.actions:
+            match = re.escape(perm_action)
+            match = re.match("(?i)^" + match.replace(r"\*", ".*") + "$", action)
+            if match:
+                return None
+
+    return f"{action} permission is missing"
+
+
+def validate_resource(client, key, resource, actions):
+    perms = client.permissions.list_for_resource(resource['resource_group'],
+                                                 resource['namespace'],
+                                                 "",
+                                                 resource['type'],
+                                                 resource['name'])
+    errors = []
+    for action in actions:
+        perms, perms_copy = tee(perms)
+        perms_list = list(perms_copy)
+        error = can_do_action(perms_list, action)
+        if error is not None:
+            row = [key, resource['name'], error]
+            errors.append(row)
+
+    return errors
+
+
+def get_subnet(client, subnet, subnet_parts):
+    try:
+        subnet_obj = client.subnets.get(subnet_parts['resource_group'],
+                                        subnet_parts['name'],
+                                        subnet_parts['child_name_1'])
+    except Exception as err:
+        if isinstance(err, ResourceNotFoundError):
```

Comment:
I believe you can simply use `except ResourceNotFoundError as err` and then another `except` underneath without needing `isinstance`